# (PART) PUBLICATION STATS {-}

## Creating objects for search terms
```{r}
tic()

year <- 1991:2022
microbiome_search <- glue("microbiome OR metagenomics OR metagenome AND {year}[PDAT]")
genomics_search <- glue("genomics AND {year}[PDAT]")
proteomics_search <- glue("proteomics AND {year}[PDAT]")
metabolomics_search <- glue("metabolomics AND {year}[PDAT]")
lipidomics_search <- glue("lipidomics AND {year}[PDAT]")
phenomics_search <- glue("phenomics AND {year}[PDAT]")
transcriptomics_search <- glue("transcriptomics AND {year}[PDAT]")
all_search <- glue("{year}[PDAT]")
```


## Getting search counts
```{r}
omics_search_counts <- tibble(year = year,
       microbiome_search = microbiome_search,
       genomics_search = genomics_search,
       proteomics_search = proteomics_search,
       metabolomics_search = metabolomics_search,
       lipidomics_search = lipidomics_search,
       phenomics_search = phenomics_search,
       transcriptomics_search = transcriptomics_search,
       all_search = all_search) %>%
  mutate(
         microb_res = map_dbl(microbiome_search, ~entrez_search(db="pubmed", term=.x)$count),
         genom_res = map_dbl(genomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         proteo_res = map_dbl(proteomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         metabol_res = map_dbl(metabolomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         lipid_res = map_dbl(lipidomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         pheno_res = map_dbl(phenomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         transcr_res = map_dbl(transcriptomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         all_res = map_dbl(all_search, ~entrez_search(db="pubmed", term=.x)$count)
         ) 

# save results in an R object
omics_search_counts %>% 
  saveRDS("data/omics_search_counts.rds")
```


## Plotting the results 
```{r}
readRDS("data/omics_search_counts.rds") %>% 
  filter(year >= 1991 ) %>% 
  mutate(
         Microbiome = microb_res / all_res,
         Genomics = genom_res / all_res,
         Proteomics = proteo_res / all_res,
         Metabolomics = metabol_res / all_res,
         Lipidomics = lipid_res / all_res,
         Phenomics = pheno_res / all_res,
         Transcriptomics = transcr_res / all_res) %>%
  select(year, Microbiome, Genomics, Proteomics, Metabolomics, Lipidomics, Phenomics, Transcriptomics) %>%
  pivot_longer(-year) %>%
  filter(value > 0) %>% 
  
  ggplot(aes(x = year, y = value, group = name, color = name)) +
  geom_line(size = 1) +

  scale_color_manual(name = "Omics Type",
                     values = c("blue2", "maroon", "red","orange", "green", "magenta", "purple")) +
  labs(x = "Year", y = "Percentage of all Papers in PubMed") +
  theme_classic() 

ggsave("figures/omics_compare.png", width = 6, height = 4)

toc() # time to run
```

