# (PART) TEXT SUMMARIZATION {-}

# Extract Sentences from an Article

## Article summarization
```{r}
library(tidyverse)
## Abstract

text <- "Rapid progress in the development of next-generation sequencing (NGS) technologies in recent years has providedmany valuable insights into complex biological systems, ranging from cancer genomics to diverse microbialcommunities. NGS-based technologies for genomics, transcriptomics, and epigenomics are now increasingly focusedon the characterization of individual cells. These single-cell analyses will allow researchers to uncover new andpotentially unexpected biological discoveries relative to traditional profiling methods that assess bulk populations.Single-cell RNA sequencing (scRNA-seq), for example, can reveal complex and rare cell populations, uncover regulatoryrelationships between genes, and track the trajectories of distinct cell lineages in development. In this review, we willfocus on technical challenges in single-cell isolation and library preparation and on computational analysis pipelinesavailable for analyzing scRNA-seq data. Further technical improvements at the level of molecular and cell biology andin available bioinformatics tools will greatly facilitate both the basic science and medical applications of thesesequencing technologies."
  
abstract  <- lexRankr::lexRank(text,
                docId = rep(1, length(text)),
                n = 5,
                continuous = TRUE) %>%
dplyr::mutate(sentenceId = as.numeric(stringr::str_remove_all(sentenceId, ".*_"))) %>%
mutate(value =  round(100 * value, digits = 4),
       section = "abstract") %>% 
select(section, value, sentence) %>% 
dplyr::arrange(-value)


## Introduction

text <- "Rapid progress in the development of next-generation sequencing (NGS) technologies in recent years has providedmany valuable insights into complex biological systems, ranging from cancer genomics to diverse microbialcommunities. NGS-based technologies for genomics, transcriptomics, and epigenomics are now increasingly focusedon the characterization of individual cells. These single-cell analyses will allow researchers to uncover new andpotentially unexpected biological discoveries relative to traditional profiling methods that assess bulk populations.Single-cell RNA sequencing (scRNA-seq), for example, can reveal complex and rare cell populations, uncover regulatoryrelationships between genes, and track the trajectories of distinct cell lineages in development. In this review, we willfocus on technical challenges in single-cell isolation and library preparation and on computational analysis pipelinesavailable for analyzing scRNA-seq data. Further technical improvements at the level of molecular and cell biology andin available bioinformatics tools will greatly facilitate both the basic science and medical applications of thesesequencing technologies."
  
intro  <- lexRankr::lexRank(text,
                docId = rep(1, length(text)),
                n = 3,
                continuous = TRUE) %>%
dplyr::mutate(sentenceId = as.numeric(stringr::str_remove_all(sentenceId, ".*_"))) %>%
       mutate(value =  round(100 * value, digits = 4),
              section = "intro") %>% 
       select(section, value, sentence) %>% 
dplyr::arrange(-value)


## Research and Sequencing Strategy

text <- "Rapid progress in the development of next-generation sequencing (NGS) technologies in recent years has providedmany valuable insights into complex biological systems, ranging from cancer genomics to diverse microbialcommunities. NGS-based technologies for genomics, transcriptomics, and epigenomics are now increasingly focusedon the characterization of individual cells. These single-cell analyses will allow researchers to uncover new andpotentially unexpected biological discoveries relative to traditional profiling methods that assess bulk populations.Single-cell RNA sequencing (scRNA-seq), for example, can reveal complex and rare cell populations, uncover regulatoryrelationships between genes, and track the trajectories of distinct cell lineages in development. In this review, we willfocus on technical challenges in single-cell isolation and library preparation and on computational analysis pipelinesavailable for analyzing scRNA-seq data. Further technical improvements at the level of molecular and cell biology andin available bioinformatics tools will greatly facilitate both the basic science and medical applications of thesesequencing technologies."

## Rank sentences
methods <- lexRankr::lexRank(text,
                  docId = rep(1, length(text)),
                  n = 2,
                  continuous = TRUE) %>%
  dplyr::mutate(sentenceId = as.numeric(stringr::str_remove_all(sentenceId, ".*_"))) %>%
       mutate(value =  round(100 * value, digits = 4),
              section = "methods") %>% 
       select(section, value, sentence) %>% 
  dplyr::arrange(-value)


## Results 

text <- "Rapid progress in the development of next-generation sequencing (NGS) technologies in recent years has providedmany valuable insights into complex biological systems, ranging from cancer genomics to diverse microbialcommunities. NGS-based technologies for genomics, transcriptomics, and epigenomics are now increasingly focusedon the characterization of individual cells. These single-cell analyses will allow researchers to uncover new andpotentially unexpected biological discoveries relative to traditional profiling methods that assess bulk populations.Single-cell RNA sequencing (scRNA-seq), for example, can reveal complex and rare cell populations, uncover regulatoryrelationships between genes, and track the trajectories of distinct cell lineages in development. In this review, we willfocus on technical challenges in single-cell isolation and library preparation and on computational analysis pipelinesavailable for analyzing scRNA-seq data. Further technical improvements at the level of molecular and cell biology andin available bioinformatics tools will greatly facilitate both the basic science and medical applications of thesesequencing technologies."

## Rank sentences
results <- lexRankr::lexRank(text,
                  docId = rep(1, length(text)),
                  n = 20,
                  continuous = TRUE) %>%
  dplyr::mutate(sentenceId = as.numeric(stringr::str_remove_all(sentenceId, ".*_"))) %>%
       mutate(value =  round(100 * value, digits = 4),
              section = "results") %>% 
       select(section, value, sentence) %>% 
  dplyr::arrange(-value)


## Conclusion

text <- "Rapid progress in the development of next-generation sequencing (NGS) technologies in recent years has providedmany valuable insights into complex biological systems, ranging from cancer genomics to diverse microbialcommunities. NGS-based technologies for genomics, transcriptomics, and epigenomics are now increasingly focusedon the characterization of individual cells. These single-cell analyses will allow researchers to uncover new andpotentially unexpected biological discoveries relative to traditional profiling methods that assess bulk populations.Single-cell RNA sequencing (scRNA-seq), for example, can reveal complex and rare cell populations, uncover regulatoryrelationships between genes, and track the trajectories of distinct cell lineages in development. In this review, we willfocus on technical challenges in single-cell isolation and library preparation and on computational analysis pipelinesavailable for analyzing scRNA-seq data. Further technical improvements at the level of molecular and cell biology andin available bioinformatics tools will greatly facilitate both the basic science and medical applications of thesesequencing technologies."

## Rank sentences
if(length(text) > 1) {
  conclusion  <- lexRankr::lexRank(text,
                  docId = 1,
                  n = 2,
                  continuous = TRUE) %>%
  dplyr::mutate(sentenceId = as.numeric(stringr::str_remove_all(sentenceId, ".*_"))) %>%
       mutate(value =  round(100 * value, digits = 4),
              section = "conclusion") %>% 
       select(section, value, sentence) %>% 
  dplyr::arrange(-value)
}else{
  conclusion <- text %>% as.data.frame() %>% 
    mutate(value =100, .before=1) 
    colnames(conclusion)[2] <- "sentence"
    
    conclusion <- conclusion %>% 
      mutate(section = "conclusion") %>% 
       select(section, value, sentence)
}


## Reference

## Merge annotation

annotation <- ""
annotation <- rbind(annotation, abstract, intro, methods,  results, conclusion) %>% 
  as.data.frame()
saveRDS(annotation, "data/annotation.rds")

```