# (PART) TEXT MINING {-}

# Extract Sentences from Article

```{r message=FALSE, warning=FALSE, include=FALSE}
# General Setup

options(stringsAsFactors = F)          # no automatic data transformation
options("scipen" = 100, "digits" = 12) # suppress math annotation

source(file = "R/common.R")
if(!require("xml2")) {install.packages("xml2")}
library(xml2)
if(!require("rvest")) {install.packages("rvest")}
library(rvest)
if(!require("lexRankr")) {install.packages("lexRankr")}
library(lexRankr)
if(!require("textmineR")) {install.packages("textmineR")}
library(textmineR)
if(!require("tidyverse")) {install.packages("tidyverse")}
library(tidyverse, suppressPackageStartupMessages())
if(!require("quanteda")) {install.packages("quanteda")}
library(quanteda)
if(!require("igraph")) {install.packages("igraph")}
library(igraph)
if(!require("here")) {install.packages("here")}
library(here)
if(!require("remotes")) {install.packages("remotes")}
library(remotes)
if(!require("klippy")) {remotes::install_github("rlesur/klippy")}
library(klippy)

klippy::klippy()

# Text summarization
## Path to article

# url = "https://complexdatainsights.com/projects/ngs-flyer/pmid_34171432.html"
# 
# ## Extract sentences
# # read page html
# page = xml2::read_html(url)
# # page = xml2::read_html(url)
# 
# # extract text from page html using selector
# page %>%
#   # extract paragraphs
#   rvest::html_nodes("p") %>%
#   # extract text
#   rvest::html_text() %>%
#   # remove empty elements
#   .[. != ""] -> text
# # inspect data
# # head(text, 2)
```

## Using Text
```{r eval=FALSE, include=FALSE, echo=T}
library(tidyverse)
## Abstract

text <- "Add text here"
  
abstract  <- lexRankr::lexRank(text,
                docId = rep(1, length(text)),
                n = 5,
                continuous = TRUE) %>%
dplyr::mutate(sentenceId = as.numeric(stringr::str_remove_all(sentenceId, ".*_"))) %>%
mutate(value =  round(100 * value, digits = 4),
       section = "abstract") %>% 
select(section, value, sentence) %>% 
dplyr::arrange(-value)


## Introduction

text <- "Add text here"
  
intro  <- lexRankr::lexRank(text,
                docId = rep(1, length(text)),
                n = 3,
                continuous = TRUE) %>%
dplyr::mutate(sentenceId = as.numeric(stringr::str_remove_all(sentenceId, ".*_"))) %>%
       mutate(value =  round(100 * value, digits = 4),
              section = "intro") %>% 
       select(section, value, sentence) %>% 
dplyr::arrange(-value)


## Research and Sequencing Strategy

text <- "Add text here"

## Rank sentences
methods <- lexRankr::lexRank(text,
                  docId = rep(1, length(text)),
                  n = 2,
                  continuous = TRUE) %>%
  dplyr::mutate(sentenceId = as.numeric(stringr::str_remove_all(sentenceId, ".*_"))) %>%
       mutate(value =  round(100 * value, digits = 4),
              section = "methods") %>% 
       select(section, value, sentence) %>% 
  dplyr::arrange(-value)


## Results 

text <- "Add text here"

## Rank sentences
results <- lexRankr::lexRank(text,
                  docId = rep(1, length(text)),
                  n = 20,
                  continuous = TRUE) %>%
  dplyr::mutate(sentenceId = as.numeric(stringr::str_remove_all(sentenceId, ".*_"))) %>%
       mutate(value =  round(100 * value, digits = 4),
              section = "results") %>% 
       select(section, value, sentence) %>% 
  dplyr::arrange(-value)


## Conclusion

text <- "Add text here"

## Rank sentences
if(length(text) > 1) {
  conclusion  <- lexRankr::lexRank(text,
                  docId = 1,
                  n = 2,
                  continuous = TRUE) %>%
  dplyr::mutate(sentenceId = as.numeric(stringr::str_remove_all(sentenceId, ".*_"))) %>%
       mutate(value =  round(100 * value, digits = 4),
              section = "conclusion") %>% 
       select(section, value, sentence) %>% 
  dplyr::arrange(-value)
}else{
  conclusion <- text %>% as.data.frame() %>% 
    mutate(value =100, .before=1) 
    colnames(conclusion)[2] <- "sentence"
    
    conclusion <- conclusion %>% 
      mutate(section = "conclusion") %>% 
       select(section, value, sentence)
}


## Reference

#Refer [@]

## Merge annotation

annotation <- ""
annotation <- rbind(annotation, abstract, intro, methods,  results, conclusion) %>% 
  as.data.frame()
saveRDS(annotation, "RDataRDS/annotation.rds")

```

# Example 1: PMID: 34171432
**Single-cell RNA-sequencing atlas reveals an MDK-dependent immunosuppressive environment in ErbB pathway-mutated gallbladder cancer**

## Introduction
Gallbladder carcinoma (GBC) is the most common cancer of the biliary tract and the sixth most common type of gastrointestinal cancer worldwide. Growing evidence suggests that genomic alterations acquired during oncogenesis help tumor cells to escape immune surveillance. ErbB signaling is the most extensively mutated pathway that mediates anti-tumor immunity. This study took advantage of the newly created scRNA-seq technology to test the hypothesis that individual cellular components of the tumor microenvironment (TME) in GBC function differentially to participate in ErbB pathway mutation-dependent tumor progression.

## Research Strategy
- This study engaged single-cell RNA-sequencing to reveal transcriptomic heterogeneity and intercellular crosstalk from 13 human GBCs and adjacent normal tissues.
- WES analysis was performed to reveal the genomic variations related to tumor malignancy.
- A variety of bulk RNA-sequencing, immunohistochemical staining, immunofluorescence staining and functional experiments were employed to study the difference between tissues with or without ErbB pathway mutations. See Figure 1.

<br>
<br>


```{r strategy, fig.cap="Research Strategy"}
library(DiagrammeR)
library(DiagrammeRsvg)

mermaidgraph <- mermaid("graph TD
A[DISCOVERY COHORT] --> |GBC Patients| B[Ball Bladder<br>Fresh Biopsy Samples]
subgraph SEQUENCING & PRELIMINARY ANALYSIS</strong>
B[Gallbladder Fresh Biopsy Samples] --> C[Cell Suspension]
C --> D[Single-Cell <br>RNA-sequencing]
D --> E[Transcriptomic <br>Heterogeneity Analysis]
B --> F[Tissue Fragments]
F --> G[Bulk <br>RNA-Sequencing]
G --> H[RNA-Sequencing<br>Relation Analysis]
F --> I[Whole Exome <br>Sequencing]
I --> J[Mutation Analysis]
end
E --> K{ErbB Pathway Mutation}
H --> K
J --> K
subgraph INTERCELLULAR CROSSTALK ANALYSIS
K --> L[Fibroblast]
K --> M[Macrophage]
K --> N[CD8+ T]
K --> O[CD4+ T]
K --> P[Treg]
K -.-> Q[Others]
end

", height = 800, width = 1000)

mermaidgraph
```

<br>
<br>

## Results
- Tumors with ErbB pathway mutations harbored a larger population of subtype 1 and 2 epithelial cells, Tregs, and M2 macrophages.
- Increased MDK in these tumors interacted with its receptor LRP1 to promote immunosuppressive macrophage differentiation.
- Crosstalk between macrophage-secreted CXCL10 and its receptor CXCR3 on Tregs was induced in GBC with ErbB pathway mutation.


## Conclusion
This study provided valuable insights into transcriptomic heterogeneity and the global cellular network in the TME, which coordinately functions to promote the progression of GBC with ErbB pathway mutations; thus, unveiling novel cellular and molecular targets for cancer therapy.


## References
Primary reference, [@Zhang2021]

# Example 2: PMID: 30902578
**Drosophila Histone Demethylase KDM5 Regulates Social Behavior through Immune Control and Gut Microbiota Maintenance**

## Introduction
The gut microbiota can affect the symptoms of intellectual disability (ID) and autism spectrum disorder (ASD) diseases. Loss of function mutations in histone demethylases KDM5A, KDM5B, or KDM5C are found in patients with ID and ASD. However, the molecular mechanisms that regulate host-commensal microbiota homeostasis in normal and disease states remain largely unknown. Drosophila is a widely accepted model for studying behavior.


## Research Strategy
This study used *Drosophila melanogaster* as a model organism to investigate the mechanistic contribution of KDM5 to social behavior by regulating the gut microbiome composition. Research strategy major workflows include:

- A: EXPERIMENTAL MODEL AND SUBJECT DETAILS
- B: METHOD DESCRIPTION
- C: IMMUNE DEFICIENT PATHWAY
- D: MICROBIOME, QUANTIFICATION, TRANSCRIPTOMICS
- E: SOCIAL BEHAVIOR

> Please, see attached `Figures` folder for graphical workflows.


<!-- - A: EXPERIMENTAL MODEL AND SUBJECT DETAILS -->
  <!-- - Fly Husbandry and Fly Stocks -->
  <!-- - S2 Cell Line  -->
<!-- - B: METHOD DESCRIPTION -->
  <!-- - KDM5 and/ or IMD Knockdown in Intestine Tissue at the Adult Stage -->
  <!-- - Germ-Free Flies and Re-colonization of Germ-Free Flies -->
  <!-- - 16S rRNA Sequencing and Analysis -->
  <!-- - Quantification of Bacterial Species by q-PCR -->
  <!-- - RNA Extraction, cDNA Library Preparation, RNA-Seq Sequencing -->
  <!-- - Bacterial Culture and Colonies Analysis -->
  <!-- - Assessment of Consumed Food Volumes -->
  <!-- - Intestine Integrity Analysis -->
  <!-- - Feeding Assay with FITC-Labeled Beads -->
  <!-- - Chromatin Immunoprecipitation -->
  <!-- - Functional Interpretation of Deferentially Expressed Genes and Gene Network Analysis -->
  <!-- - Immunofluorescence -->
  <!-- - L. plantarum Identification and Feeding -->
  <!-- - Providencia rettgeri Identification -->
  <!-- - Social Space Analysis -->
  <!-- - Direct Social Contacting Analysis -->
  <!-- - Social Avoidance Analysis of Drosophila Stress Odorant -->
  <!-- - Lifespan Assays -->
  <!-- - Serotonin Detection -->
  <!-- - Transmission Electron Microscopy -->
  <!-- - Gram-Staining of Gut Microbiota -->
  <!-- - Western Blot -->
  <!-- - Real-Time PCR -->
  <!-- - Plasmid Construction -->
  <!-- - Transfection -->
  <!-- - Inducing kdm5 RNAi in S2 Cells by Transfection with dsRNA -->
  <!-- - S2 Cells with 20E and PGN Treatment -->
  <!-- - Agarose Gel Electrophoresis for Checking Open Reading Frame Expression of PGRP-LC Isforms -->
<!-- - QUANTIFICATION AND STATISTICAL ANALYSIS -->


```{r combo_reserach_strategy, fig.cap="Figure 1: Research Strategy"}
library(DiagrammeR)
library(DiagrammeRsvg)

mermaid("graph TD
subgraph A: EXPERIMENTAL MODEL AND SUBJECT DETAILS</strong>
A[Model Organism: <i>Drosophilla melanogaster</i>] --> B[Fly Husbandry & Stocks]
B --> C1[On Standard Medium]
B --> D1[Wild Type]
B --> E1[S2 Cell Line]
end

subgraph B: EXPERIMENTAL METHODS DETAILS
C1[On Standard Medium] --> F{Adult Flies}
D1[Wild Type] --> F
F --> F1[Intestinal Tissue]
F1 --> G[KDM5 &/or IMD Knockdown]
G --> H[Generate Germ-Free Flies]
H --> I[16S rRNA Gene PCR Analysis]
I --> |Transfer on germ-free food| J[Re-colonization of Germ-Free Flies]
J --> K5[Assessment of Consumed Food Volumes]
K5[Assessment of Consumed Food Volumes] --> K6[Intestine Integrity Analysis]
K6[Intestine Integrity Analysis] --> K7[Feeding Assay with FITC-Labeled Beads]
end

subgraph C: MICROBIOME, QUANTIFICATION, TRANSCRIPTOMICS
J --> F2{GF-Intestine Tissue}
GF{Germ-Free Adult Flies} --> F2
GF --> LA[Lifespan Assays]
GF --> SD[Serotonin Detection]
GF --> TEM[Transmission Electron Microscopy]
F2 -->  |Gut Microbiome| K1[16S rRNA Sequencing and Analysis]
F2 --> |Bacterial detection| K2[Quantification of Bacterial <br> Species by q-PCR]
F2 --> |Transcriptomics| K3[RNA Extraction <br> cDNA Library Preparation <br> RNA-Seq Sequencing]
F2 --> |Bacterial Culture| K4[Bacterial Culture and Colonies Analysis]
F2 --> |Protein-DNA Enrichment| K8[Chromatin Immunoprecipitation]
K8[Chromatin Immunoprecipitation] --> |Cellular Component<br>Biological Process<br>Molecular Function| K9[Functional Interpretation of <br>Deferentially Expressed Genes <br>and Gene Network Analysis] 
F2 --> |Fluorescent Imaging| K10[Immunofluorescence]
F2 --> |Isolation or Culturing<br>Genomic DNA Extraction<br> 16S rRNA Amplification | K11[<i>L. plantarum</i> Identification and Feeding <br> <i>Providencia rettgeri</i> Identification]
F2 --> |Feature Observation| K12[Transmission Electron Microscopy<br> Gram-Staining of Gut Microbiota]
F2 --> |Protein Detection<br>Quantification<br>Recombinant DNA| K13[Western Blot<br>Real-Time PCR<br>Plasmid Construction]
end

K1[16S rRNA Sequencing and Analysis] --> L{QUANTIFICATION AND <br> STATISTICAL ANALYSIS}
K2[Quantification of Bacterial Species <br> by q-PCR] --> L
K3[RNA Extraction, cDNA Library Preparation, <br>  RNA-Seq Sequencing] --> L
K4[Bacterial Culture and Colonies Analysis] --> L
K9[Functional Interpretation of <br>Deferentially Expressed Genes <br>and Gene Network Analysis] --> L
K10[Immunofluorescence] --> L
K11[<i>L. plantarum</i> Identification and Feeding <br> <i>Providencia rettgeri</i> Identification] --> L
K12[Transmission Electron Microscopy<br> Gram-Staining of Gut Microbiota<br>] --> L
K13[Western Blot<br>Real-Time PCR<br>Plasmid Construction] --> L

subgraph D: SOCIAL BEHAVIOR
GF --> SB1[Social Space Analysis]
SB1 --> SB2[Direct Social Contacting Analysis]
SB1 --> SB3[Social Avoidance Analysis of Drosophila Stress Odorant]
end

subgraph E: IMMUNE DEFICIENT PATHWAY
E1[S2 Cell Line] --> |Cultured| TF1[Transfection]
TF1[Transfection] --> |Induce kdm5 RNAi| TF2[Transfection with dsRNA]
TF2[Transfection with dsRNA] --> |20E and PGN Treatment| IMD[Activate IMD Pathway]
IMD[Activate IMD Pathway] --> |Agarose Gel Electrophoresis| ORF[Checking Open Reading Frame <br> Expression of PGRP-LC Isforms]
end
", height = 800, width = 1000)
```

```{r reserach_strategy, fig.cap="Figure 1: Research Strategy", include=FALSE}
library(DiagrammeR)
library(DiagrammeRsvg)

mermaid("graph TD
subgraph A: EXPERIMENTAL MODEL AND SUBJECT DETAILS</strong>
A[Model Organism: <i>Drosophilla melanogaster</i>] --> B[Fly Husbandry & Stocks]
B --> C1[On Standard Medium]
B --> D1[Wild Type]
B --> E1[S2 Cell Line]
end", height = 800, width = 1000)

mermaid("graph TD

subgraph B: EXPERIMENTAL METHODS DETAILS
C1[On Standard Medium] --> F{Adult Flies}
D1[Wild Type] --> F{Adult Flies}
F{Adult Flies} --> F1[Intestinal Tissue]
F1[Intestinal Tissue] --> G[KDM5 &/or IMD Knockdown]
G[KDM5 &/or IMD Knockdown] --> H[Generate Germ-Free Flies]
H[Generate Germ-Free Flies] --> I[16S rRNA Gene PCR Analysis]
I[16S rRNA Gene PCR Analysis] --> |Transfer on germ-free food| J[Re-colonization of Germ-Free Flies]
J[Re-colonization of Germ-Free Flies] --> K5[Assessment of Consumed Food Volumes]
K5[Assessment of Consumed Food Volumes] --> K6[Intestine Integrity Analysis]
K6[Intestine Integrity Analysis] --> K7[Feeding Assay with FITC-Labeled Beads]
end", height = 800, width = 1000)

mermaid("graph TD

subgraph C: MICROBIOME, QUANTIFICATION, TRANSCRIPTOMICS
J[Re-colonization of Germ-Free Flies] --> F2[GF-Intestine Tissue]
GF{Germ-Free Adult Flies} --> F2
GF --> LA[Lifespan Assays]
GF --> SD[Serotonin Detection]
GF --> TEM[Transmission Electron Microscopy]
F2 --> |Gut Microbiome| K1[16S rRNA Sequencing and Analysis]
F2 --> |Bacterial detection| K2[Quantification of Bacterial <br> Species by q-PCR]
F2 --> |Transcriptomics| K3[RNA Extraction <br> cDNA Library Preparation <br> RNA-Seq Sequencing]
F2 --> |Bacterial Culture| K4[Bacterial Culture and Colonies Analysis]
F2 --> |Protein-DNA Enrichment| K8[Chromatin Immunoprecipitation]
K8 --> |Cellular Component<br>Biological Process<br>Molecular Function| K9[Functional Interpretation of <br>Deferentially Expressed Genes <br>and Gene Network Analysis] 
F2 --> |Fluorescent Imaging| K10[Immunofluorescence]
F2 --> |Isolation or Culturing<br>Genomic DNA Extraction<br> 16S rRNA Amplification | K11[<i>L. plantarum</i> Identification and Feeding <br> <i>Providencia rettgeri</i> Identification]
F2 --> |Feature Observation| K12[Transmission Electron Microscopy<br> Gram-Staining of Gut Microbiota]
F2 --> |Protein Detection<br>Quantification<br>Recombinant DNA| K13[Western Blot<br>Real-Time PCR<br>Plasmid Construction]
end

K1[16S rRNA Sequencing and Analysis] --> L{QUANTIFICATION AND <br> STATISTICAL ANALYSIS}
K2[Quantification of Bacterial Species <br> by q-PCR] --> L
K3[RNA Extraction, cDNA Library Preparation, <br>  RNA-Seq Sequencing] --> L
K4[Bacterial Culture and Colonies Analysis] --> L
K9[Functional Interpretation of <br>Deferentially Expressed Genes <br>and Gene Network Analysis] --> L
K10[Immunofluorescence] --> L
K11[<i>L. plantarum</i> Identification and Feeding <br> <i>Providencia rettgeri</i> Identification] --> L
K12[Transmission Electron Microscopy<br> Gram-Staining of Gut Microbiota<br>] --> L
K13[Western Blot<br>Real-Time PCR<br>Plasmid Construction] --> L
", height = 800, width = 1000)

mermaid("graph TD
subgraph D: SOCIAL BEHAVIOR
GF{Germ-Free Adult Flies} --> SB1[Social Space Analysis]
SB1 --> SB2[Direct Social Contacting Analysis]
SB1 --> SB3[Social Avoidance Analysis of Drosophila Stress Odorant]
end
", height = 800, width = 1000)

mermaid("graph TD
subgraph E: IMMUNE DEFICIENT PATHWAY
E1[S2 Cell Line] --> |Cultured| TF1[Transfection]
TF1 --> |Induce kdm5 RNAi| TF2[Transfection with dsRNA]
TF2 --> |20E and PGN Treatment| IMD[Activate IMD Pathway]
IMD --> |Agarose Gel Electrophoresis| ORF[Checking Open Reading Frame <br> Expression of PGRP-LC Isforms]
end", height = 800, width = 1000)

# mermaidgraph

```


<br>

## Results
- Reducing KDM5 causes intestinal barrier dysfunction and changes in social behavior that correlates with compositional changes in the gut microbiota.
- Therapeutic alteration of the dysbiotic microbiota through antibiotic administration or feeding with a probiotic Lactobacillus strain partially rescues the behavioral, lifespan, and cellular phenotypes observed in kdm5-deficient flies.
- KDM5 demethylase activity was found to transcriptionally
regulate component genes of the immune deficiency (IMD) signaling pathway and gut-microbiome-brain function.


## Conclusion
This research suggests that modification of the gut microbiome can serve as a clinical therapeutic approach for ID and ASD patients with aberrant IMD signaling.



## References
Primary reference, [@Chen2019]



# Example 3: PMID: 33106631
**Triticum population sequencing provides insights into wheat adaptation** 

## Introduction
Bread wheat has expanded its habitat from
a small core area to a broad spectrum of diverse environments around the globe. The genetic mechanisms of the evolutionary success of wheat adaptation are not well understood. This study used wheat as a model and performed a genus-level whole-genome sequencing to get insights into the genetic driving forces behind bread wheat adaptation.

## Research Strategy Workflow
- Worldwide collection of wheat accessions.
- Sequencing and quality control.
- Variant calling.
- Phylogenetic analysis.
- Nucleotide diversity analysis.
- Population-based introgression analysis.
- Identification of wheat orthologs.
- Genomic signatures of convergent adaptation.

Figure 1: Research Strategy (On a separate file)


```{r triticum_reserach_strategy, fig.cap="Figure 1: Research Strategy"}
library(DiagrammeR)
library(DiagrammeRsvg)

mermaid("graph TD
subgraph A: COLLECTION OF WHEAT ACCESSION
A(Worldwide Collection of Wheat Accessions) --> B(West Asia<br>Africa<br>Europe<br>East Asia<br>South Asia<br>Central Asia<br>America)
B --> C[Triticum genera]
C --> D[Illumina NGS Sequencing]
D --> |Platform &<br>Seq Depth| E[HiSeq 2500 & NovaSeq 6000, Average Depth 3.8X - 10X]
E --> F[Reads Quality Control]
end 

subgraph B: VARIANT IDENTIFICATION AND QUALITY CONTROL
F --> G((Variant Calling))
G--> QC{Quality Control of SNPs}
QC --> H1[Quality Filter]
QC --> H2[Depth Filter]
QC --> H3[Segregation Test Filter]
QC --> H4[Linkage Disequilibrium Filter]
QC --> H5[IBD: Identical-By-Descent Filter]
QC --> H6[Minor Allele Count Filter]
H1--> G1((HIGH QUALITY SNPs))
H2--> G1
H3--> G1
H4--> G1
H5--> G1
H6--> G1
end
subgraph C: MULTIPLE ANALYSES TO GAIN INSIGHTS INTO SINGLE NUCLEOTIDE POLYMORPHISMS
G1 --> I[Detection of <br>Syntenic Sites]
G1 --> J[Construction of VMap <br> by Merging SNPs <br>from Taxonomic Groups]
G1 --> K[Identification of <br>Ancestral Allele]
G1 --> L1[Phylogenetic <br>Analysis]
G1 --> L2[Genetic Diversity<br> Analysis]
G1 --> L3[Population-Based<br>Introgression Analysis]
G1 --> L4[Identification of<br> Wheat Orthologs]
G1 --> L5[Detect Genomic Signatures <br> of Convergent Adaptation ]
end

", height = 800, width = 1000)
```


## Results
- Whole-genome sequencing of populations of subspecies within the <i>Triticum</i> and <i>Aegilops</i> genera, enabled the identification of composite introgression from wild populations contributing to the genetic diversity of the bread wheat genome, allowing its divergent adaptation. 
- Convergent adaptation to human selection showed 2- to 16-fold enrichment relative to random expectation.
- A certain set of genes were repeatedly selected in Triticum species despite their drastic differences in ploidy levels and growing zones, indicating the important role of evolutionary constraints in shaping the adaptive landscape of bread wheat.


## Conclusion
A genus-level whole-genome sequencing study of bread wheat provided insights into the genetic driving forces behind wheat adaptation and evolutionary constraints of wheat adaptation to human selection. These results showed the genetic necessities of wheat as a global crop and provided new perspectives on transferring adaptive success across species for crop improvement.

Primary reference, [@Zhou2020]

## References



