<!-- # <strong>Article to review</strong>: Single-Cell RNA-Sequencing Atlas Reveals an MDK-Dependent Immunosuppressive Environment in ErbB Pathway-Mutated Gallbladder Cancer  -->

```{r include=FALSE}
source("R/common.R")
library(flexdashboard)
library(tidyverse, suppressPackageStartupMessages())
```

PART 1 {data-width=350}
-----------------------------------------------------------------------

### Introduction

```{r eval=FALSE, include=FALSE}
readRDS("RDataRDS/annotation.rds") %>% 
  filter(section == "intro")
```
- Gallbladder carcinoma (GBC) is the most common cancer of the biliary tract and the sixth most common type of gastrointestinal cancer worldwide.
- Growing evidence suggests that genomic alterations acquired during oncogenesis help tumor cells to escape immune surveillance.
- ErbB signaling is the most extensively mutated pathway that mediates anti-tumor immunity
- This study tested the hypothesis that individual cellular components of the tumor microenvironment (TME) in GBC function differentially to participate in ErbB pathway mutation-dependent tumor progression.

[Primary reference ](https://doi.org/10.1016/J.JHEP.2021.06.023), [@Zhang2021]


### Research Strategy
```{r include=FALSE}
readRDS("RDataRDS/annotation.rds") %>% 
  filter(section == "methods")
```

- This study engaged single-cell RNA-sequencing to reveal transcriptomic heterogeneity and intercellular crosstalk from 13 human GBCs and adjacent normal tissues. 
- WES analysis was performed to reveal the genomic variations related to tumor malignancy. 
- A variety of bulk RNA-sequencing, immunohistochemical staining, immunofluorescence staining and functional experiments were employed to study the difference between tissues with or without ErbB pathway mutations.

```{r include=FALSE}
library(DiagrammeR)
library(DiagrammeRsvg)

graph <- DiagrammeR::grViz("
digraph {
# graph [layout = circo, rankdir = TD]
graph [rankdir = TD]

node [
shape = box, 
style = filled, 
fillcolor = white, 
fontname = Helvetica,
penwidth = 2.0] 

edge [arrowhead = diamond]

A [label = 'Discovery cohort\nGBCPatients']
B [label = 'Gallbladder\nFresh Biopsy Samples']
C [label = 'Cell Suspension']
D [label = 'Single-Cell \nRNA-sequencing']
E [label = 'Transcriptomic \nHeterogeneity Analysis']
F [label = 'Tissue Fragments']
G [label = 'Bulk \nRNA-Sequencing']
H [label = 'RNA-Sequencing\nRelation Analysis']
I [label = 'Whole Exome \nSequencing']
J [label = 'Mutation Analysis']
K [label = 'ErbB Pathway\nMutation']
L [label = 'Fibroblast']
M [label = 'Macrophage']
N [label = 'CD8+ T']
O [label = 'CD4+ T']
P [label = 'Treg']
Q [label = 'Others']

{A} -> B
{B} -> C
{C} -> D
{D} -> E
{B} -> F
{F} -> G
{G} -> H
{F} -> I
{I} -> J
{E} -> K
{H} -> K
{J} -> K
{K} -> L
{K} -> M
{K} -> N
{K} -> O
{K} -> P
{K} -> Q


}", height = 500, width = 500)

graph

# 2. Convert to SVG, then save as png
strategy = DiagrammeRsvg::export_svg(graph)
strategy = charToRaw(strategy) # flatten
rsvg::rsvg_png(strategy, "img/strategy.png")
```


```{r mermaid, include=FALSE}
library(DiagrammeR)
library(DiagrammeRsvg)

graph <- mermaid("graph TD
A[DISCOVERY COHORT] --> |GBC Patients| B[Ball Bladder<br>Fresh Biopsy Samples]
subgraph SEQUENCING & PRELIMINARY ANALYSIS</strong>
B[Ball Bladder Fresh Biopsy Samples] --> C[Cell Suspension]
C --> D[Single-Cell <br>RNA-sequencing]
D --> E[Transcriptomic <br>Heterogeneity Analysis]
B --> F[Tissue Fragments]
F --> G[Bulk <br>RNA-Sequencing]
G --> H[RNA-Sequencing<br>Relation Analysis]
F --> I[Whole Exome <br>Sequencing]
I --> J[Mutation Analysis]
end
E --> K{ErbB Pathway Mutation}
H --> K
J --> K
subgraph INTERCELLULAR CROSSTALK ANALYSIS
K --> L[Fibroblast]
K --> M[Macrophage]
K --> N[CD8+ T]
K --> O[CD4+ T]
K --> P[Treg]
K -.-> Q[Others]
end

", height = 800, width = 1000)
  # export_graph(graph,
  #   file_name = "img/mermaidgraph",
  #   file_type = "PNG")
```



PART 2 {data-width=450}
-----------------------------------------------------------------------
```{r include=FALSE}
graph
```


<!-- ![Workflow of experimental strategy](img/workflow.png){width=70%} -->
<a href=""><img src="img/workflow.png" alt="flowchart" width=80%" style="margin: 0px 10% 0px 10%; float: middle;"/>Workflow of experimental strategy</a>


```{r eval=FALSE, include=FALSE}
DiagrammeR::mermaid("
graph TD
A[Christmas] -->|Get money| B(Go shopping)
subgraph Nerve wracking
B --> C{Roll a dice}
end
subgraph Don't look
  C -->|One| D[Laptop]
  C -->|Two| E[iPhone]
  subgraph High Probability
    C-->|Three| G
    C-->|Four| G
    C-->|Five| G[A gift card]
  end
  subgraph Low Probability
    C -->|Six| F[fa:fa-car Car]
  end
end
")
```


<!-- ![](img/strategy.png) -->

PART 3 {data-width=350}
-----------------------------------------------------------------------


### Results

```{r include=FALSE}
readRDS("RDataRDS/annotation.rds") %>% 
  filter(section == "results")
```
- Tumors with ErbB pathway mutations harbored a larger population of subtype 1 and 2 epithelial cells, Tregs, and M2 macrophages.
- Increased MDK in these tumors interacted with its receptor LRP1 to promote immunosuppressive macrophage differentiation.
- Crosstalk between macrophage-secreted CXCL10 and its receptor CXCR3 on Tregs was induced in GBC with ErbB pathway mutation.



```{r magick_magrittr_gif, fig.height=6, fig.width=8, include=FALSE}
### Graphical Results

library(magick)
library(magrittr)
list.files(path="images/", pattern = 'Screen.*.png', full.names = TRUE) %>% 
        image_read() %>% 
        image_join() %>%
        image_animate(fps=0.5) %>% 
        image_write("./images/mdk_gbc.gif") 
```

```{r gifski_gif, eval=FALSE, fig.height=6, fig.width=8, include=FALSE}
library(gifski)
png_files <- list.files("images", pattern = "Screen.*.png", full.names = TRUE)
gifski(png_files, gif_file = "./images/mdk_gbc.gif", width = 800, height = 600, delay = 2)
```

 
<!-- <a href=""><img src="images/mdk_gbc.gif" alt="mdk_gbc_gif" width=80%" style="margin: 0px 10% 0px 10%; float: middle;"/></a> -->


### Conclusion

```{r include=FALSE}
readRDS("RDataRDS/annotation.rds") %>% 
  filter(section == "conclusion")
```

This study provided valuable insights into transcriptomic heterogeneity and the global cellular network in the TME, which coordinately functions to promote the progression of GBC with ErbB pathway mutations; thus, unveiling novel cellular and molecular targets for cancer therapy.

<br>

### References

```{r eval=FALSE, include=FALSE}
## Word count
paste("Total number of words for Intro, Methods, Results and Conclusion, respectively")

intro <- "- Gallbladder carcinoma (GBC) is the most common cancer of the biliary tract and the sixth most common type of gastrointestinal cancer worldwide.
- Growing evidence suggests that genomic alterations acquired during oncogenesis help tumor cells to escape immune surveillance.
- ErbB signaling is the most extensively mutated pathway that mediates anti-tumor immunity
- This study tested the hypothesis that individual cellular components of the tumor microenvironment (TME) in GBC function differentially to participate in ErbB pathway mutation-dependent tumor progression."

lengths(gregexpr("\\W+", intro)) + 1


methods <- "- This study engaged single-cell RNA-sequencing to reveal transcriptomic heterogeneity and intercellular crosstalk from 13 human GBCs and adjacent normal tissues. 
- WES analysis was performed to reveal the genomic variations related to tumor malignancy. 
- A variety of bulk RNA-sequencing, immunohistochemical staining, immunofluorescence staining and functional experiments were employed to study the difference between tissues with or without ErbB pathway mutations."

sum((lengths(gregexpr("\\W+", methods)) + 1))

results <- "
- Tumors with ErbB pathway mutations harbored a larger population of subtype 1 and 2 epithelial cells, Tregs, and M2 macrophages.
- ErbB pathway mutations resulted in immunosuppressive macrophage differentiation and regulatory T cell activation, explaining the reduced anti-cancer immunity and worse overall survival observed in patients with these mutations.
- Increased MDK in these tumors interacted with its receptor LRP1 to promote immunosuppressive macrophage differentiation.
- Crosstalk between macrophage-secreted CXCL10 and its receptor CXCR3 on Tregs was induced in GBC with ErbB pathway mutation."
sum((lengths(gregexpr("\\W+", results)) + 1))

conclusion <- "This study has provided valuable insights into transcriptomic heterogeneity and the global cellular network in the TME, which coordinately functions to promote the progression of GBC with ErbB pathway mutations; thus, unveiling novel cellular and molecular targets for cancer therapy"
sum((lengths(gregexpr("\\W+", conclusion)) + 1))

paste("Total= ", 
    
    (sum((lengths(gregexpr("\\W+", intro)) + 1), 
    (lengths(gregexpr("\\W+", methods)) + 1), 
    (lengths(gregexpr("\\W+", results)) + 1),
    (lengths(gregexpr("\\W+", conclusion)) + 1))))
```



