---
title: "PubMed Article Searching"
subtitle: "Practical user guides with integrated solutions"
author: "Teresia Mrema-Buza, A Bioinformatics Scientist and Data Science Enthusiast"
date: "Updated on `r Sys.Date()`"
site: bookdown::bookdown_site
knit: bookdown::render_book
output:
  bookdown::bs4_book:
    includes:
      in_header:
    template: bs4_template.html
        
  bookdown::gitbook:
    includes:
      in_header: header.html
    config: 
      toc:
       collapse: 
      search: yes
      fontsettings:
        size: 2
    split_by: section
    split_bib: no
    df_print: paged
    number_sections: true
    
  html_document:
    code_folding: hide
    
  bookdown::html_document2:
    code_folding: hide
    
  bookdown::pdf_book:
    config:
      toc = true
      toc_depth = 2
      number_sections = true
      fig_caption = true
      keep_tex = no
      pandoc_args = null
      toc_unnumbered = no
      toc_appendix = no
      toc_bib = no 
      quote_footer = null
      highlight_bw = true
      latex_engine = xelatex
      df_print = kable
      base_format = rmarkdown::pdf_document
  includes:
    in_header: 
    - latex/header.tex
    - latex/preamble.tex
documentclass: book
classoption: openany #remove empty pages in pdf doc
bibliography:
- library/book.bib
- library/packages.bib
- library/bibliograph.bib
- library/software.bib 
citation_package:
- natbib
- biblatex
- amsplain
colorlinks: true
css:
  - css/style.css
  - style.css
always_allow_html: true
fig_caption: true
fontsize: 12pt
geometry: margin=1in
indent: false
keep_tex: true
link-citations: true
mainfont: Times New Roman
biblio-style: apalike
spacing: double
header-includes: 
- \usepackage{setspace}
- \newpage
- \newenvironment{tmbinfo}[0]{}{}
- \renewenvironment{tmbinfo}[0]{}{}
- \newenvironment{tmbalert}[0]{}{}
- \renewenvironment{tmbalert}[0]{}{}
- \newenvironment{tmbshare}[0]{}{}
- \renewenvironment{tmbshare}[0]{}{}
description: |
  This guide provides integrated and highly curated solutions for achieving better results.
---


```{r pkgbiblib, include=FALSE}
# automatically create a bib database for R packages
knitr::write_bib(c(.packages(), 'base','bookdown','rmarkdown','tidyverse'), 'library/packages.bib')
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo = TRUE,
	message = FALSE,
	warning = FALSE
)

## Create Object Folder 
if (!dir.exists("RDataRDS")) {dir.create("RDataRDS")}

# source(file = "R/common.R")

htmltools::tagList(rmarkdown::html_dependency_font_awesome())

knitr::opts_chunk$set(
  echo = TRUE,
  message = FALSE,
  warning = FALSE,
  cache = FALSE,
  comment = NA,
  fig.path='./figures/',
  fig.show='asis',
  dev = 'png',
  fig.align='center',
  out.width = "80%",
  out.height = "40%",
  fig.width = 7,
  fig.asp = 0.618,
  fig.show = "asis"
)
```

# Getting Started with Literature Mining {-#frontpage}

<br>

<a href=""><img src="img/cover.gif" alt="Book cover" width="100%" style="padding: 50px 0px 50px 0px; float: right;"/></a>

<br>

## Quick Glimpse {-}

<br>

## License {-}

![](global/CCbyNCND.png){ width=100px } The online version of this book is free and licensed under a <a rel="license" href="http://creativecommons.org/licenses/by-nc-nd/4.0/">Creative Commons Attribution-NonCommercial-NoDerivatives 4.0 International License</a>. The code that built this guide is available at a private GitHub repository. A minimal code and quick description of the book is available at a public [GitHub Page](https://tmbuza.github.io/Systematic-Microbiome-Data-Analysis/). If interested you can obtain a more detailed copy from the [CDI LLC Store](https://complexdatainsights.com/products).

## Structure of this guide {-}
This guide is divided into chapters to facilitate easy navigation. Each chapter contains several sections as displayed in the navigation bars on the left and right. Click the hyper-linked text if you want to jump into a specific chapter or section.

<!--chapter:end:index.Rmd-->

# (PART) PUBMED TRACKING {-}

# Searching and Tracking PubMed

## Manual download from PubMed 
```{r}
library(tidyverse)

tumorlympho <- read_csv("data/Cancer_PubMed_Timeline_Results_by_Year.csv",
         skip=1, show_col_types = F) %>% 
  filter(Year != 2023) %>% 
  mutate(Omics = "TumorLympho")

microbiome <- read_csv("../Datasets/microbiome_metagenomic_PubMed_Timeline.csv",
         skip=1, show_col_types = F) %>% 
  filter(Year != 2023) %>% 
  mutate(Omics = "Microbiome")

genomics <- read_csv("../Datasets/genomics_PubMed_Timeline.csv",
         skip=1, show_col_types = F) %>% 
  filter(Year != 2023) %>% 
  mutate(Omics = "Genomics")

metabolomics <- read_csv("../Datasets/metabolomics_PubMed_Timeline.csv",
         skip=1, show_col_types = F) %>% 
  filter(Year != 2023) %>% 
  mutate(Omics = "Metabolomics")

proteomics <- read_csv("../Datasets/proteomics_PubMed_Timeline.csv",
         skip=1, show_col_types = F) %>% 
  filter(Year != 2023) %>% 
  mutate(Omics = "Proteomics")

phenomics <- read_csv("../Datasets/phenomics_PubMed_Timeline.csv",
         skip=1, show_col_types = F) %>% 
  filter(Year != 2023) %>% 
  mutate(Omics = "Phenomics")

lipidomics <- read_csv("../Datasets/lipidomics_PubMed_Timeline.csv",
         skip=1, show_col_types = F) %>% 
  filter(Year != 2023) %>% 
  mutate(Omics = "Lipidomics")

transcriptomics <- read_csv("../Datasets/transcriptomics_PubMed_Timeline.csv",
         skip=1, show_col_types = F) %>% 
  filter(Year != 2023) %>% 
  mutate(Omics = "Transcriptomics")

do.call("rbind", list(tumorlympho, microbiome, genomics, metabolomics, proteomics, phenomics, lipidomics, transcriptomics)) %>% 
  write_csv("../Datasets/omics_pubmed_timeline.csv")

read_csv("../Datasets/omics_pubmed_timeline.csv") %>% 
  ggplot(aes(x=Year, y=Count, color = Omics)) +
  geom_line() +
  theme_classic()
```


## Using `rentrez` package
```{r eval=FALSE, echo=TRUE}
library(rentrez)
library(glue)

year <- 1991:2022
tumorlympho_search <- glue("cancer OR tumor AND lymphocyte AND {year}[PDAT]")
microbiome_search <- glue("microbiome OR metagenomics OR metagenome AND {year}[PDAT]")
genomics_search <- glue("genomics AND {year}[PDAT]")
proteomics_search <- glue("proteomics AND {year}[PDAT]")
metabolomics_search <- glue("metabolomics AND {year}[PDAT]")
lipidomics_search <- glue("lipidomics AND {year}[PDAT]")
phenomics_search <- glue("phenomics AND {year}[PDAT]")
transcriptomics_search <- glue("transcriptomics AND {year}[PDAT]")

all_search <- glue("{year}[PDAT]")

omics_search_counts <- tibble(year = year,
       tumorlympho_search = tumorlympho_search,
       microbiome_search = microbiome_search,
       genomics_search = genomics_search,
       proteomics_search = proteomics_search,
       metabolomics_search = metabolomics_search,
       lipidomics_search = lipidomics_search,
       phenomics_search = phenomics_search,
       transcriptomics_search = transcriptomics_search,
       all_search = all_search) %>%
  mutate(tumlymph_res = map_dbl(tumorlympho_search, ~entrez_search(db="pubmed", term=.x)$count),
         microb_res = map_dbl(microbiome_search, ~entrez_search(db="pubmed", term=.x)$count),
         genom_res = map_dbl(genomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         proteo_res = map_dbl(proteomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         metabol_res = map_dbl(metabolomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         lipid_res = map_dbl(lipidomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         pheno_res = map_dbl(phenomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         transcr_res = map_dbl(transcriptomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         all = map_dbl(all_search, ~entrez_search(db="pubmed", term=.x)$count)
         ) 

omics_search_counts %>% 
  saveRDS("../Datasets/omics_search_counts.rds")
```

## Compare Omics publications
```{r}
readRDS("../Datasets/omics_search_counts.rds") %>% 
  filter(year >= 1991 ) %>% 
  mutate(TumorLympho = tumlymph_res / all,
         Microbiome = microb_res / all,
         Genomics = genom_res / all,
         Proteomics = proteo_res / all,
         Metabolomics = metabol_res / all,
         Lipidomics = lipid_res / all,
         Phenomics = pheno_res / all,
         Transcriptomics = transcr_res / all) %>%
  select(year, TumorLympho, Microbiome, Genomics, Proteomics, Metabolomics, Lipidomics, Phenomics, Transcriptomics) %>%
  pivot_longer(-year) %>%
  filter(value > 0) %>% 
  ggplot(aes(x=year, y=value, group=name, color=name)) +
  geom_line(size=1) +
  scale_y_log10(limits=c(NA, 100),
                breaks=c(0, 0.1, 1, 10, 100),
                labels=c("0", "0.1", "1", "10", "100")) +
  # scale_y_continuous(labels = scales::percent)
  scale_color_manual(name="Omics Type",
                     values=c("blue2", "maroon", "red","orange", "green","blue", "magenta", "purple")) +
  labs(x="Year", y="Percentage of all Papers in PubMed") +
  theme_classic() 

ggsave("figures/omics_compare.png", width=6, height=4)
```


<!--chapter:end:01_omics_pubmed_timeline.Rmd-->

# (APPENDIX) APPENDIX {-}

# Saved Data Objects

```{r include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```


## RDS Format for Individual Object
```{r}

rds <- list.files(path="./RDataRDS", pattern = "*.rds", full.names = TRUE)
rds

```

## RData Format for Multiple Objects
```{r}

rdata <- list.files(path="./RDataRDS", pattern = "*.RData", full.names = TRUE)
rdata

```


## CSV or TSV Format Files
```{r}

list.files(path="./RDataRDS", pattern = "*.csv|tsv", full.names = TRUE)

```


```{r eval=FALSE, include=FALSE}
## All-in-One Input-Output Data
lsdata("RDataRDS/saved_objects.RData")
```



## How to reload RDS or RData
```{block, type="tmbinfo", echo=T}
### RDS format e.g. foo.rds
foo <-`readRDS("RDataRDS/foo.rds")`

### RData format e.g. foo.RData
`load("RDataRDS/foo.RData", verbose = TRUE)`

### List objects in RData
`lsdata("foo.RData")`
```



# Software and Packages

## Basic dependencies
* `r R.version.string`
* `tidyverse` (v. `r packageVersion("tidyverse")`)
* `knitr` (v. `r packageVersion("knitr")`)
* `rmarkdown` (v. `r packageVersion("rmarkdown")`)
* `bookdown` (v. `r packageVersion("bookdown")`)


## Available on machine used 
```{r}
sessionInfo()

```




<!--chapter:end:99_appendix.Rmd-->

`r if (knitr:::is_html_output()) '
# References {-}
'`


<!--chapter:end:999-references.Rmd-->

