---
title: "Text Mining in R"
subtitle: "Searching and visually comparing different fields"
author: "Teresia Mrema-Buza"
date: "<strong>Last Updated</strong>: `r Sys.Date()`"
site: bookdown::bookdown_site
knit: bookdown::render_book
output: 
  flexdashboard::flex_dashboard:
    orientation: columns
    # vertical_layout: fill
    # vertical_layout: fill
    vertical_layout: scroll
  # html_document:
  # pdf_document:
  # word_document:
  bookdown::bs4_book:
    template: bs4_text_mining_template.html
    toc_depth: 6
    favicon: "favicon.ico"
    footnotes_inline: false
    split_bib: true
    repo:
      base: https://github.com/DataInsights/text-mining/
      branch: main
      # subdir: book
      icon: "fas fa-air-freshener"
    theme:
      fg: "black"
      bg: "#FFFEFA"
      primary: "#277DB0"
      secondary: "#277DB0"
      success: "#277DB0"
      info: "#277DB0"
      warning: "#277DB0"
      danger: "#277DB0"
      base_font:
        google: Sen
      heading_font:
        google:
          family: Bitter
          wght: 300
      code_font:
        google:
          # arguments to sass::font_google()
          family: DM Mono
          local: false
css:
- style.css
bibliography:
- library/articles.bib
- library/software.bib
citation_package:
- natbib
- biblatex
- amsplain
---

```{r setup, include=FALSE}
source("R/common.R")
library(flexdashboard)
library(tidyverse, suppressPackageStartupMessages())
```


# Pubmed Literature Tracking in R
```{r pubmed, fig.cap="PubMed Search Workflow"}
library(DiagrammeR)
library(DiagrammeRsvg)

mermaidgraph <- mermaid("graph TD
A(OMICS PUBLICATION STATISTICS) --> A0{NCBI PubMed Database}
A0 --> A1[Microbiome OR Metabolomics]
A0 --> A2[Genomics]
A0 --> A3[Proteomics]
A0 --> A4[Metabolomics]
A0 --> A5[Phenomics]
A0 --> A6[Transcriptomics]


subgraph ARTICLE SEARCH USING OMICS KEYWORDS
A1 --> B{Indexed Papers in PubMed}
A2 --> B
A3 --> B
A4 --> B
A5 --> B
A6 --> B

B --> |Download results| C[Results by Year in CSV format]
end

subgraph SEARCH RESULTS
C --> |Filename| D[PubMed_Timeline_Results_by_Year.csv]
D --> E[Save file in data directory]
E -.-> F[Further Analysis]
end

", height = 600, width = 600)

mermaidgraph

```




<!--chapter:end:index.Rmd-->

# (PART) PUBMED TRACKING {-}

# Searching and Tracking PubMed

## Manual download from PubMed 
```{r}
library(tidyverse)

microbiome <- read_csv("../Datasets/microbiome_metagenomic_PubMed_Timeline.csv",
         skip=1, show_col_types = F) %>% 
  filter(Year != 2023) %>% 
  mutate(Omics = "Microbiome")

genomics <- read_csv("../Datasets/genomics_PubMed_Timeline.csv",
         skip=1, show_col_types = F) %>% 
  filter(Year != 2023) %>% 
  mutate(Omics = "Genomics")

metabolomics <- read_csv("../Datasets/metabolomics_PubMed_Timeline.csv",
         skip=1, show_col_types = F) %>% 
  filter(Year != 2023) %>% 
  mutate(Omics = "Metabolomics")

proteomics <- read_csv("../Datasets/proteomics_PubMed_Timeline.csv",
         skip=1, show_col_types = F) %>% 
  filter(Year != 2023) %>% 
  mutate(Omics = "Proteomics")

phenomics <- read_csv("../Datasets/phenomics_PubMed_Timeline.csv",
         skip=1, show_col_types = F) %>% 
  filter(Year != 2023) %>% 
  mutate(Omics = "Phenomics")

lipidomics <- read_csv("../Datasets/lipidomics_PubMed_Timeline.csv",
         skip=1, show_col_types = F) %>% 
  filter(Year != 2023) %>% 
  mutate(Omics = "Lipidomics")

transcriptomics <- read_csv("../Datasets/transcriptomics_PubMed_Timeline.csv",
         skip=1, show_col_types = F) %>% 
  filter(Year != 2023) %>% 
  mutate(Omics = "Transcriptomics")

do.call("rbind", list(microbiome, genomics, metabolomics, proteomics, phenomics, lipidomics, transcriptomics)) %>% 
  write_csv("../Datasets/omics_pubmed_timeline.csv")

read_csv("../Datasets/omics_pubmed_timeline.csv") %>% 
  ggplot(aes(x=Year, y=Count, color = Omics)) +
  geom_line() +
  theme_classic()
```


## Using `rentrez` package
```{r eval=FALSE, include=FALSE}
library(rentrez)
library(glue)

year <- 1991:2022
microbiome_search <- glue("microbiome OR metagenomics OR metagenome AND {year}[PDAT]")
genomics_search <- glue("genomics AND {year}[PDAT]")
proteomics_search <- glue("proteomics AND {year}[PDAT]")
metabolomics_search <- glue("metabolomics AND {year}[PDAT]")
lipidomics_search <- glue("lipidomics AND {year}[PDAT]")
phenomics_search <- glue("phenomics AND {year}[PDAT]")
transcriptomics_search <- glue("transcriptomics AND {year}[PDAT]")

all_search <- glue("{year}[PDAT]")

omics_search_counts <- tibble(year = year,
       microbiome_search = microbiome_search,
       genomics_search = genomics_search,
       proteomics_search = proteomics_search,
       metabolomics_search = metabolomics_search,
       lipidomics_search = lipidomics_search,
       phenomics_search = phenomics_search,
       transcriptomics_search = transcriptomics_search,
       all_search = all_search) %>%
  mutate(microb_res = map_dbl(microbiome_search, ~entrez_search(db="pubmed", term=.x)$count),
         genom_res = map_dbl(genomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         proteo_res = map_dbl(proteomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         metabol_res = map_dbl(metabolomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         lipid_res = map_dbl(lipidomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         pheno_res = map_dbl(phenomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         transcr_res = map_dbl(transcriptomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         all = map_dbl(all_search, ~entrez_search(db="pubmed", term=.x)$count)
         ) 

omics_search_counts %>% 
  saveRDS("../Datasets/omics_search_counts.rds")
```

## Compare Omics publications
```{r}
readRDS("../Datasets/omics_search_counts.rds") %>% 
  filter(year >= 1991 ) %>% 
  mutate(Microbiome = 100 * microb_res / all,
         Genomics = 100 * genom_res / all,
         Proteomics = 100 * proteo_res / all,
         Metabolomics = 100 * metabol_res / all,
         Lipidomics = 100 * lipid_res / all,
         Phenomics = 100 * pheno_res / all,
         Transcriptomics = 100 * transcr_res / all) %>%
  select(year, Microbiome, Genomics, Proteomics, Metabolomics, Lipidomics, Phenomics, Transcriptomics) %>%
  pivot_longer(-year) %>%
  filter(value > 0) %>% 
  ggplot(aes(x=year, y=value, group=name, color=name)) +
  geom_line(size=1) +
  scale_y_log10(limits=c(NA, 100),
                breaks=c(0, 0.1, 1, 10, 100),
                labels=c("0", "0.1", "1", "10", "100")) +
  labs(x="Year", y="Percentage of all OMICS\n Papers in PubMed") +
  scale_color_manual(name="Omics Type",
                     values=c("maroon", "red","orange", "green","blue", "magenta", "purple")) +
  labs(x="Year", y="Published Papers in PubMed (%)", color = "OMICS") +
  theme_classic() 

ggsave("figures/omics_compare.png", width=6, height=4)
```


<!--chapter:end:01_omics_pubmed_timeline.Rmd-->

# (APPENDIX) APPENDIX {-}

# Saved Data Objects

```{r include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```


## RDS Format for Individual Object
```{r}

rds <- list.files(path="./RDataRDS", pattern = "*.rds", full.names = TRUE)
rds

```

## RData Format for Multiple Objects
```{r}

rdata <- list.files(path="./RDataRDS", pattern = "*.RData", full.names = TRUE)
rdata

```


## CSV or TSV Format Files
```{r}

list.files(path="./RDataRDS", pattern = "*.csv|tsv", full.names = TRUE)

```


```{r eval=FALSE, include=FALSE}
## All-in-One Input-Output Data
lsdata("RDataRDS/saved_objects.RData")
```



## How to reload RDS or RData
```{block, type="tmbinfo", echo=T}
### RDS format e.g. foo.rds
foo <-`readRDS("RDataRDS/foo.rds")`

### RData format e.g. foo.RData
`load("RDataRDS/foo.RData", verbose = TRUE)`

### List objects in RData
`lsdata("foo.RData")`
```



# Software and Packages

## Basic dependencies
* `r R.version.string`
* `tidyverse` (v. `r packageVersion("tidyverse")`)
* `knitr` (v. `r packageVersion("knitr")`)
* `rmarkdown` (v. `r packageVersion("rmarkdown")`)
* `bookdown` (v. `r packageVersion("bookdown")`)


## Available on machine used 
```{r}
sessionInfo()

```




<!--chapter:end:99_appendix.Rmd-->

`r if (knitr:::is_html_output()) '
# References {-}
'`


<!--chapter:end:999-references.Rmd-->

