---
title: "PubMed Article Searching in R"
subtitle: "A simple practical user guides for publication statistics"
author: "Teresia Mrema-Buza, A Bioinformatics and Data Science Enthusiast"
date: "Updated on `r Sys.Date()`"
site: bookdown::bookdown_site
knit: bookdown::render_book
output:
  bookdown::bs4_book:
    includes:
      in_header:
    template: bs4_template.html
        
  bookdown::gitbook:
    includes:
      in_header: header.html
    config: 
      toc:
       collapse: 
      search: yes
      fontsettings:
        size: 2
    split_by: section
    split_bib: no
    df_print: paged
    number_sections: true
documentclass: book
classoption: openany
bibliography:
- library/packages.bib
- library/bibliograph.bib
citation_package:
- natbib
- biblatex
- amsplain
colorlinks: true
css:
  - style.css
always_allow_html: true
fig_caption: true
fontsize: 12pt
geometry: margin=1in
indent: false
keep_tex: true
link-citations: true
mainfont: Times New Roman
biblio-style: apalike
spacing: double
description: |
  This guide provides integrated and highly curated solutions for achieving better results.
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo = TRUE,
	message = FALSE,
	warning = FALSE
)

## Create Object Folder 
if (!dir.exists("library")) {dir.create("library")}
if (!dir.exists("data")) {dir.create("data")}
if (!file.exists("library/bibliograph.bib")) {file.create("library/bibliograph.bib")}

# source(file = "R/common.R")

htmltools::tagList(rmarkdown::html_dependency_font_awesome())

knitr::opts_chunk$set(
  echo = FALSE,
  message = FALSE,
  warning = FALSE,
  cache = FALSE,
  comment = NA,
  fig.path='./figures/',
  fig.show='asis',
  dev = 'png',
  fig.align='center',
  out.width = "80%",
  out.height = "40%",
  fig.width = 7,
  fig.asp = 0.618,
  fig.show = "asis"
)
```

# Getting Started withSearching PubMed Articles {#pubmed-search}

```{r pkgbiblib, include=FALSE}
knitr::write_bib(c(.packages(), 'base','bookdown','rmarkdown','tidyverse'), 'library/packages.bib')
```


## Load libraries
```{r}
library(rentrez)
library(glue)
library(tidyverse, suppressPackageStartupMessages)
library(DiagrammeR)
library(DiagrammeRsvg)
library(tictoc)
```

## Manual search workflow
```{r workflow}
mermaidgraph <- mermaid("graph TD

subgraph SEARCHING PUBMED DATABASE
A[NCBI PubMed Database<br>https://pubmed.ncbi.nlm.nih.gov/] --> B[Search Term]
B --> |Search Results| C[Indexed Papers in PubMed]
end
subgraph SEARCH RESULTS
C --> |Download results| D{Results by Year in CSV format}
D --> |Filename| E[PubMed_Timeline_Results_by_Year.csv]
E --> F[Save file in data directory]
end

F -.-> G[Further Analysis in R or Python]

", height = 500, width = 300)

mermaidgraph

```


## Code availability {-}
The code that generated this guide is available at a public [GitHub repository](https://github.com/DataInsights/pubmed-tracking/) or via a [GH-Page](https://DataInsights.github.io/pubmed-tracking/).





<!--chapter:end:index.Rmd-->

# (PART) PUBLICATION STATS {-}

## Creating objects for search terms
```{r}
tic()

year <- 1991:2022
microbiome_search <- glue("microbiome OR metagenomics OR metagenome AND {year}[PDAT]")
genomics_search <- glue("genomics AND {year}[PDAT]")
proteomics_search <- glue("proteomics AND {year}[PDAT]")
metabolomics_search <- glue("metabolomics AND {year}[PDAT]")
lipidomics_search <- glue("lipidomics AND {year}[PDAT]")
phenomics_search <- glue("phenomics AND {year}[PDAT]")
transcriptomics_search <- glue("transcriptomics AND {year}[PDAT]")
all_search <- glue("{year}[PDAT]")
```


## Getting search counts
```{r}
omics_search_counts <- tibble(year = year,
       microbiome_search = microbiome_search,
       genomics_search = genomics_search,
       proteomics_search = proteomics_search,
       metabolomics_search = metabolomics_search,
       lipidomics_search = lipidomics_search,
       phenomics_search = phenomics_search,
       transcriptomics_search = transcriptomics_search,
       all_search = all_search) %>%
  mutate(
         microb_res = map_dbl(microbiome_search, ~entrez_search(db="pubmed", term=.x)$count),
         genom_res = map_dbl(genomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         proteo_res = map_dbl(proteomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         metabol_res = map_dbl(metabolomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         lipid_res = map_dbl(lipidomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         pheno_res = map_dbl(phenomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         transcr_res = map_dbl(transcriptomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         all_res = map_dbl(all_search, ~entrez_search(db="pubmed", term=.x)$count)
         ) 

# save results in an R object
omics_search_counts %>% 
  saveRDS("data/omics_search_counts.rds")
```


## Plotting the results 
```{r}
readRDS("data/omics_search_counts.rds") %>% 
  filter(year >= 1991 ) %>% 
  mutate(
         Microbiome = microb_res / all_res,
         Genomics = genom_res / all_res,
         Proteomics = proteo_res / all_res,
         Metabolomics = metabol_res / all_res,
         Lipidomics = lipid_res / all_res,
         Phenomics = pheno_res / all_res,
         Transcriptomics = transcr_res / all_res) %>%
  select(year, Microbiome, Genomics, Proteomics, Metabolomics, Lipidomics, Phenomics, Transcriptomics) %>%
  pivot_longer(-year) %>%
  filter(value > 0) %>% 
  
  ggplot(aes(x = year, y = value, group = name, color = name)) +
  geom_line(size = 1) +

  scale_color_manual(name = "Omics Type",
                     values = c("blue2", "maroon", "red","orange", "green", "magenta", "purple")) +
  labs(x = "Year", y = "Percentage of all Papers in PubMed") +
  theme_classic() 

ggsave("figures/omics_compare.png", width = 6, height = 4)

toc() # time to run
```


<!--chapter:end:01_omics_pubmed_timeline.Rmd-->

# (PART) TEXT SUMMARIZATION {-}

# Extract Sentences from an Article

## Article summarization
```{r}
library(tidyverse)
## Abstract

text <- "Rapid progress in the development of next-generation sequencing (NGS) technologies in recent years has providedmany valuable insights into complex biological systems, ranging from cancer genomics to diverse microbialcommunities. NGS-based technologies for genomics, transcriptomics, and epigenomics are now increasingly focusedon the characterization of individual cells. These single-cell analyses will allow researchers to uncover new andpotentially unexpected biological discoveries relative to traditional profiling methods that assess bulk populations.Single-cell RNA sequencing (scRNA-seq), for example, can reveal complex and rare cell populations, uncover regulatoryrelationships between genes, and track the trajectories of distinct cell lineages in development. In this review, we willfocus on technical challenges in single-cell isolation and library preparation and on computational analysis pipelinesavailable for analyzing scRNA-seq data. Further technical improvements at the level of molecular and cell biology andin available bioinformatics tools will greatly facilitate both the basic science and medical applications of thesesequencing technologies."
  
abstract  <- lexRankr::lexRank(text,
                docId = rep(1, length(text)),
                n = 5,
                continuous = TRUE) %>%
dplyr::mutate(sentenceId = as.numeric(stringr::str_remove_all(sentenceId, ".*_"))) %>%
mutate(value =  round(100 * value, digits = 4),
       section = "abstract") %>% 
select(section, value, sentence) %>% 
dplyr::arrange(-value)


## Introduction

text <- "Rapid progress in the development of next-generation sequencing (NGS) technologies in recent years has providedmany valuable insights into complex biological systems, ranging from cancer genomics to diverse microbialcommunities. NGS-based technologies for genomics, transcriptomics, and epigenomics are now increasingly focusedon the characterization of individual cells. These single-cell analyses will allow researchers to uncover new andpotentially unexpected biological discoveries relative to traditional profiling methods that assess bulk populations.Single-cell RNA sequencing (scRNA-seq), for example, can reveal complex and rare cell populations, uncover regulatoryrelationships between genes, and track the trajectories of distinct cell lineages in development. In this review, we willfocus on technical challenges in single-cell isolation and library preparation and on computational analysis pipelinesavailable for analyzing scRNA-seq data. Further technical improvements at the level of molecular and cell biology andin available bioinformatics tools will greatly facilitate both the basic science and medical applications of thesesequencing technologies."
  
intro  <- lexRankr::lexRank(text,
                docId = rep(1, length(text)),
                n = 3,
                continuous = TRUE) %>%
dplyr::mutate(sentenceId = as.numeric(stringr::str_remove_all(sentenceId, ".*_"))) %>%
       mutate(value =  round(100 * value, digits = 4),
              section = "intro") %>% 
       select(section, value, sentence) %>% 
dplyr::arrange(-value)


## Research and Sequencing Strategy

text <- "Rapid progress in the development of next-generation sequencing (NGS) technologies in recent years has providedmany valuable insights into complex biological systems, ranging from cancer genomics to diverse microbialcommunities. NGS-based technologies for genomics, transcriptomics, and epigenomics are now increasingly focusedon the characterization of individual cells. These single-cell analyses will allow researchers to uncover new andpotentially unexpected biological discoveries relative to traditional profiling methods that assess bulk populations.Single-cell RNA sequencing (scRNA-seq), for example, can reveal complex and rare cell populations, uncover regulatoryrelationships between genes, and track the trajectories of distinct cell lineages in development. In this review, we willfocus on technical challenges in single-cell isolation and library preparation and on computational analysis pipelinesavailable for analyzing scRNA-seq data. Further technical improvements at the level of molecular and cell biology andin available bioinformatics tools will greatly facilitate both the basic science and medical applications of thesesequencing technologies."

## Rank sentences
methods <- lexRankr::lexRank(text,
                  docId = rep(1, length(text)),
                  n = 2,
                  continuous = TRUE) %>%
  dplyr::mutate(sentenceId = as.numeric(stringr::str_remove_all(sentenceId, ".*_"))) %>%
       mutate(value =  round(100 * value, digits = 4),
              section = "methods") %>% 
       select(section, value, sentence) %>% 
  dplyr::arrange(-value)


## Results 

text <- "Rapid progress in the development of next-generation sequencing (NGS) technologies in recent years has providedmany valuable insights into complex biological systems, ranging from cancer genomics to diverse microbialcommunities. NGS-based technologies for genomics, transcriptomics, and epigenomics are now increasingly focusedon the characterization of individual cells. These single-cell analyses will allow researchers to uncover new andpotentially unexpected biological discoveries relative to traditional profiling methods that assess bulk populations.Single-cell RNA sequencing (scRNA-seq), for example, can reveal complex and rare cell populations, uncover regulatoryrelationships between genes, and track the trajectories of distinct cell lineages in development. In this review, we willfocus on technical challenges in single-cell isolation and library preparation and on computational analysis pipelinesavailable for analyzing scRNA-seq data. Further technical improvements at the level of molecular and cell biology andin available bioinformatics tools will greatly facilitate both the basic science and medical applications of thesesequencing technologies."

## Rank sentences
results <- lexRankr::lexRank(text,
                  docId = rep(1, length(text)),
                  n = 20,
                  continuous = TRUE) %>%
  dplyr::mutate(sentenceId = as.numeric(stringr::str_remove_all(sentenceId, ".*_"))) %>%
       mutate(value =  round(100 * value, digits = 4),
              section = "results") %>% 
       select(section, value, sentence) %>% 
  dplyr::arrange(-value)


## Conclusion

text <- "Rapid progress in the development of next-generation sequencing (NGS) technologies in recent years has providedmany valuable insights into complex biological systems, ranging from cancer genomics to diverse microbialcommunities. NGS-based technologies for genomics, transcriptomics, and epigenomics are now increasingly focusedon the characterization of individual cells. These single-cell analyses will allow researchers to uncover new andpotentially unexpected biological discoveries relative to traditional profiling methods that assess bulk populations.Single-cell RNA sequencing (scRNA-seq), for example, can reveal complex and rare cell populations, uncover regulatoryrelationships between genes, and track the trajectories of distinct cell lineages in development. In this review, we willfocus on technical challenges in single-cell isolation and library preparation and on computational analysis pipelinesavailable for analyzing scRNA-seq data. Further technical improvements at the level of molecular and cell biology andin available bioinformatics tools will greatly facilitate both the basic science and medical applications of thesesequencing technologies."

## Rank sentences
if(length(text) > 1) {
  conclusion  <- lexRankr::lexRank(text,
                  docId = 1,
                  n = 2,
                  continuous = TRUE) %>%
  dplyr::mutate(sentenceId = as.numeric(stringr::str_remove_all(sentenceId, ".*_"))) %>%
       mutate(value =  round(100 * value, digits = 4),
              section = "conclusion") %>% 
       select(section, value, sentence) %>% 
  dplyr::arrange(-value)
}else{
  conclusion <- text %>% as.data.frame() %>% 
    mutate(value =100, .before=1) 
    colnames(conclusion)[2] <- "sentence"
    
    conclusion <- conclusion %>% 
      mutate(section = "conclusion") %>% 
       select(section, value, sentence)
}


## Reference

## Merge annotation

annotation <- ""
annotation <- rbind(annotation, abstract, intro, methods,  results, conclusion) %>% 
  as.data.frame()
saveRDS(annotation, "data/annotation.rds")

```

<!--chapter:end:02_articles_summarization.Rmd-->

# (APPENDIX) APPENDIX {-}

# Saved Data Objects

```{r include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```


## RDS Format for Individual Object
```{r}

rds <- list.files(path="./data", pattern = "*.rds", full.names = TRUE)
rds

```

## RData Format for Multiple Objects
```{r}

rdata <- list.files(path="./data", pattern = "*.RData", full.names = TRUE)
rdata

```


## CSV or TSV Format Files
```{r}

list.files(path="./data", pattern = "*.csv|tsv", full.names = TRUE)

```


```{r eval=FALSE, include=FALSE}
## All-in-One Input-Output Data
lsdata("data/saved_objects.RData")
```


## How to reload RDS or RData
```{block, type="tmbinfo", echo=T}
### RDS format e.g. foo.rds
foo <-`readRDS("data/foo.rds")`

### RData format e.g. foo.RData
`load("data/foo.RData", verbose = TRUE)`

### List objects in RData
`lsdata("foo.RData")`
```



# Software and Packages

## Basic dependencies
* `r R.version.string`
* `tidyverse` (v. `r packageVersion("tidyverse")`)
* `knitr` (v. `r packageVersion("knitr")`)
* `rmarkdown` (v. `r packageVersion("rmarkdown")`)
* `bookdown` (v. `r packageVersion("bookdown")`)

## Available on machine used 
```{r}
sessionInfo()
```




<!--chapter:end:99_appendix.Rmd-->

`r if (knitr:::is_html_output()) '
# References {-}
'`


<!--chapter:end:999-references.Rmd-->

